src/components/TradeWidget/TokenRow.tsx

```diff
@@ -145,10 +147,10 @@ const TokenRow: React.FC<Props> = ({ token, tokens, selectLabel, onSelectChange
           <strong>
             <Link to="/deposit">Exchange wallet:</Link>
           </strong>{' '}
-          <span className="success">312312.33</span>
+          <span className="success">{balance ? formatAmount(balance.exchangeBalance, balance.decimals) : 0}</span>
```

Comment:
is `balance` here always an object? if its empty (but still `{}`) it will always be truthy. Maybe better to check for `balance.exchangeBalance`

Reply:
Actually, `formatAmount` returns `null` if amount is null.
And when the wallet is disconnected the `balance` is `undefined`.

Extracted into a new function, keeping the check for `undefined` and added a check for `null` return from `formatAmount`